pslab/sciencelab.py

```diff
@@ -282,30 +282,38 @@ def _write_data_address(self, address: int, value: int):
         self.send_int(value)
         self.get_ack()
 
-    def enable_uart_passthrough(self, baudrate: int, persist=False):
+    def enable_uart_passthrough(self, baudrate: int):
         """Relay all data received by the device to TXD/RXD.
 
-        If a period > 0.5 seconds elapses between two transmit/receive events,
-        the device resets and resumes normal mode. This timeout feature has
-        been implemented in lieu of a hard reset option.
-
         Can be used to load programs into secondary microcontrollers with
         bootloaders such ATMEGA or ESP8266
 
         Parameters
         ----------
         baudrate : int
-            Baudrate of the UART bus.
-        persist : bool, optional
-            If set to True, the device will stay in passthrough mode until the
-            next power cycle. Otherwise(default scenario), the device will
-            return to normal operation if no data is sent/received for a period
-            greater than one second at a time.
+            Baudrate of the UART2 bus.
         """
+        if self.firmware.major < 3:
+            self._uart_passthrough_legacy(baudrate)
+        else:
+            self._uart_passthrough(baudrate)
+
+    def _uart_passthrough(self, baudrate: int) -> None:
```

Comment:
> The new code introduces additional complexity with more methods and conditional logic based on firmware versions. This increases the number of moving parts and duplicates logic, particularly in handling UART passthrough for different firmware versions.

The extra complexity is necessary in order to provide compatibility with different firmware versions.

> To simplify, consider refactoring to reduce duplication and encapsulate version-specific logic more concisely. For example, unify the enable_uart_passthrough method to handle both legacy and current firmware

The solution introduced in this PR of having separate methods for different firmware versions encapsulates the version-dependent logic. Unifying the `enable_uart_passthrough` method is the opposite of encapsulation.

> use a ternary operator for command selection. This will improve maintainability and reduce complexity while preserving functionality.

I disagree that a ternary operator (`command = self._uart_passthrough_legacy if self.firmware.major < 3 else self._uart_passthrough `) is more maintainable.